### PR TITLE
libextractor: 1.9 -> 1.11, add me as maintainer

### PR DESCRIPTION
**Pull request by jorsn**
* removed upstreamed patches
* enabled installCheckPhase (they work, now!)
* bumped version
* updated license (gpl2+ -> gpl3+): according to the libextractor docs,
  it was licensed under gpl3+ since version 0.7!
* added myself as maintainer

###### Motivation for this change

* updates are good: 1.9 -> 1.11
* i'm better than no maintainer
* tests are good

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after): **79 kB**
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jorsn** [comment]
I don't think the failing job has anything to do with libextractor. It's probably about pandoc and flakes (about `pandoc.x86_64-linux`, missing attr `x86_64-linux`).

File: pkgs/development/libraries/libextractor/default.nix [hunk 1]
```diff
@@ -20,28 +20,16 @@ assert gtkSupport -> glib != null && gtk3 != null;
 assert videoSupport -> ffmpeg_3 != null && libmpeg2 != null;
 
 stdenv.mkDerivation rec {
-  name = "libextractor-1.9";
+  pname = "libextractor";
+  version = "1.10";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libextractor/default.nix:25]:
```suggestion
```

done automatically by mkDerivation.

File: pkgs/development/libraries/libextractor/default.nix [hunk 2]
```diff
@@ -20,28 +20,16 @@ assert gtkSupport -> glib != null && gtk3 != null;
...
   src = fetchurl {
     url = "mirror://gnu/libextractor/${name}.tar.gz";
-    sha256 = "1zz2zvikvfibxnk1va3kgzs7djsmiqy7bmk8y01vbsf54ryjb3zh";
+    sha256 = "0mr38g7kfn3p050hd3hckbcz2yd3za6dwl1c26x2kjf7vnsi3vcy";
   };
 
-  patches = [
-    ./fix-gcc8-build.patch
-    # Fixes build with exiv2 0.27
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libextractor/default.nix:33]: Please remove fetchpatch from inputs.

File: pkgs/development/libraries/libextractor/default.nix [hunk 3]
```diff
@@ -76,10 +64,9 @@ stdenv.mkDerivation rec {
...
   # Checks need to be run after "make install", otherwise plug-ins are not in
   # the search path, etc.
-  # FIXME: Tests currently fail and the test framework appears to be deeply
-  # broken anyway.
   doCheck = false;
-  #postInstall = "make check";
+  doInstallCheck = true;
+  installCheckPhase = "make check";
 
   meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libextractor/default.nix:71]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/libraries/libextractor/default.nix [hunk 4]
```diff
@@ -105,9 +92,9 @@ stdenv.mkDerivation rec {
          additional MIME types are detected.
       '';
 
-    license = stdenv.lib.licenses.gpl2Plus;
+    license = stdenv.lib.licenses.gpl3Plus;
 
-    maintainers = [ ];
+    maintainers = [ stdenv.lib.maintainers.jorsn ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libextractor/default.nix:97]:
```suggestion
    maintainers = [ maintainers.jorsn ];
```

File: pkgs/development/libraries/libextractor/default.nix [hunk 5]
```diff
@@ -105,9 +92,9 @@ stdenv.mkDerivation rec {
          additional MIME types are detected.
       '';
 
-    license = stdenv.lib.licenses.gpl2Plus;
+    license = stdenv.lib.licenses.gpl3Plus;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libextractor/default.nix:95]:
```suggestion
    license = licenses.gpl3Plus;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by jorsn** [comment]
I did the following:
1. solved the merge conflict due to 9bb3fccb5b55326cb3c2c507464a8a28d44d1730 and 66e44425c6dfecbea68a5d6dc221ccd56561d4f1
2. incorporated all requests by @SuperSandro2000
3. Bumped it to the new (2020-01-30) version 1.11, which fixes compatibility with new exiv2
4. everything checked under *Things done* in the issue description

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 97142` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>vdrPlugins.xineliboutput</li>
</ul>
</details>
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>doodle</li>
<li>gnunet</li>
<li>gnunet-gtk</li>
<li>libextractor</li>
</ul>
</details>

**Pull request merged**